packages-ts/starknet-gauntlet/src/commands/base/executeCommand.ts

```diff
@@ -183,13 +190,28 @@ export const makeExecuteCommand = <UI, CI>(config: ExecuteCommandConfig<UI, CI>)
       await deps.prompt('Continue?')
       deps.logger.loading(`Sending transaction...`)
 
-      const tx = await this.provider.deployContract(
-        this.contract,
-        this.compiledContractHash,
-        this.input.contract,
-        false,
-        this.input?.user?.['salt'],
-      )
+      // if --classHash is included, do not re-declare contract
+      const classHash: string | undefined = this.input?.user?.['classHash']
+
+      let tx: TransactionResponse
+
+      if (classHash === undefined) {
+        tx = await this.provider.deployContract(
+          this.contract,
+          this.compiledContractHash,
+          this.input.contract,
+          false,
+          this.input?.user?.['salt'],
+        )
+      } else {
+        tx = await this.provider.onlyDeployContract(
```

Comment:
maybe this should be called `deployWithClassHash` ?

Reply:
The original starknet.js names fit best:

- `deploy`
- `declareAndDeploy`

But heads up that the new Gauntlet fork will limit a single operation to a single side-effect, and declare + deploy are two transactions.

You can check out the Starknet core extension of the new Gauntlet here: https://github.com/smartcontractkit/gauntlet-temporal-example/pull/33/files

Reply:
Maybe to clear some things up... `this.provider.deployContract` does ` declareAndDeploy which will do both and does not fail if the contract has already been deployed. `this.provider.onlyDeployContract` only does the deploy part.

You might be wondering why we need `this.provider.onlyDeployContract` if `this.provider.deployContract` fails on a re-declaration. This is because it is safer to reuse a class hash than the recompile the contract (because the contract and the compiler may have changed -- especially with how starknet updates are). 

So I think we still need 3 functions (which I'll also rename for clarity sake)

compiles the contract and declares it using the resultant generated ABI, and deploys it
`this.provider.deployContract` -> rename to `this.provider.declareAndDeployContract()`

deploys the contract given a class hash
`this.provider.onlyDeploy` -> rename to `this.provider.deployContract`

compiles the contract and declares it using the resultant generated ABI
`this.provider.declareContract` -> (no need to rename)

@krebernisak So eventually you're saying we'll need to split up the declare and deploy to two seperate Gauntlet commands?